geoapps_utils/driver/data.py

```diff
@@ -117,7 +122,17 @@ def flatten(self) -> dict:
 
         return out
 
-    @property
-    def name(self) -> str:
-        """Application name."""
-        return self._name
+    def write_ui_json(self, path: Path) -> None:
+        """
+        Write the ui.json file for the application.
+
+        :param path: Path to write the ui.json file.
+        """
+
+        if self._input_file is None:
+            ifile = InputFile.read_ui_json(self.default_ui_json, validate=False)
+        else:
+            ifile = copy(self._input_file)
+
+        ifile.data = dict(ifile.data, **self.flatten())
```

Comment:
I would need those mechanics in plate simulation to attach the dict to the UIJson group. Can we make this a method of the BaseData? Like a `get_input_file()` that makes a new instance with values updated....